### PR TITLE
Update For 1.17.1

### PR DESCRIPTION
**Pull request by ThePhilderbeast**
worldguard custom flag is broken, used OTHER_EXPLOSION instead.

### CODE REVIEW

File: src/main/java/com/philderbeast/prisonpicks/Config.java [hunk 1]
```diff
@@ -48,19 +42,26 @@
     static int DIAMOND_PRIORITY = 7;
     static int DIAMOND_BLOCK_PRIORITY = 8;
     static int EMERALD_PRIORITY = 9;
+    static int COPPER_PRIORITY = 10;
```
Comment (ThePhilderbeast) [inline src/main/java/com/philderbeast/prisonpicks/Config.java:45]: why are these priorities higher then Emeralds?

noting emeralds are the gambling currency they should probably still be the highest. and the ordering on the rest probably needs to be updated into order of value (unless the copper is higher then the value of a diamond block.

File: src/main/java/com/philderbeast/prisonpicks/Events.java [hunk 2]
```diff
@@ -56,43 +65,33 @@ public void onBlockBreak(BlockBreakEvent event)
...
-            item.setDurability(s);
+            //short s = 0;
+            
+            ((Damageable) itemMeta).setDamage(0);
+            item.setItemMeta(itemMeta);
         }
     }
-
-    @EventHandler
-    public void onBlockPlace(BlockPlaceEvent event)
```
Comment (ThePhilderbeast) [inline src/main/java/com/philderbeast/prisonpicks/Events.java:91]: why is this removed? its stops somone placing a high value block to pop the blocks around it.

File: src/main/java/com/philderbeast/prisonpicks/Pick.java [hunk 3]
```diff
@@ -23,12 +24,14 @@
     static final String FORTUNE = "fortune";
     static final String UNBREAKING = "unbreaking";
     static final String SILK_TOUCH = "silktouch";
+    
+    int rng = (int) (Math.random() * (4 - 1)) + 1;
```
Comment (ThePhilderbeast) [inline src/main/java/com/philderbeast/prisonpicks/Pick.java:28]: why is this set once for the class?????????

File: src/main/java/com/philderbeast/prisonpicks/Pick.java [hunk 4]
```diff
@@ -23,12 +24,14 @@
...
 
     public abstract void breakBlock(BlockBreakEvent event);
 
     static boolean isPick(ItemStack item)
     {
-        return item != null && item.getType() == Material.DIAMOND_PICKAXE && item.hasItemMeta() && item.getItemMeta().hasLore();
+        return item != null && item.getType() == Material.DIAMOND_PICKAXE && item.hasItemMeta() && item.getItemMeta().hasLore() || item.getType() == Material.NETHERITE_PICKAXE && item.hasItemMeta() && item.getItemMeta().hasLore();
```
Comment (ThePhilderbeast) [inline src/main/java/com/philderbeast/prisonpicks/Pick.java:34]: dont duplicate checks for the lore

File: src/main/java/com/philderbeast/prisonpicks/Pick.java [hunk 5]
```diff
@@ -116,9 +116,11 @@ void doBreak(Block block, Map<String, Boolean> enchants, Player player, Material
         }
     }
 
+    protected abstract char[] parseInt(String string);
```
Comment (ThePhilderbeast) [inline src/main/java/com/philderbeast/prisonpicks/Pick.java:119]: Class variables should all be at the top of the class

File: src/main/java/com/philderbeast/prisonpicks/Pick.java [hunk 6]
```diff
@@ -200,7 +213,7 @@ static ItemStack getDrop(int enchantmentLevel, Block block, ItemStack tool)
 
         return drop;
     }
-
+    String tpick = "";
```
Comment (ThePhilderbeast) [inline src/main/java/com/philderbeast/prisonpicks/Pick.java:216]: more random class variables?

what is this? why dosent it have an accessor label, why is it down here?

File: src/main/java/com/philderbeast/prisonpicks/Pick.java [hunk 7]
```diff
@@ -236,17 +262,41 @@ void doDamage(boolean unbreaking, Player player)
...
+                {
+                    pick = Config.EXPLOSIVE_COLOR + "Explosive Pick";
+                }else if (Pickoplenty.isPick(tool))
+                {
+                    pick = Config.PICK_O_PLENTY_COLOR + "Pick 'o' Plenty";
+                }else if (XPickoPlenty.isPick(tool))
+                {
+                    pick = Config.EXPLOSIVE_COLOR + "Explosive " + Config.PICK_O_PLENTY_COLOR + " Pick 'o' plenty";
+                }
+                Bukkit.broadcastMessage("-------------------------------------");
```
Comment (ThePhilderbeast) [inline src/main/java/com/philderbeast/prisonpicks/Pick.java:293]: why is this now a broadcast, not printed to the logs?

File: src/main/java/com/philderbeast/prisonpicks/PickCommands.java [hunk 8]
```diff
@@ -41,6 +46,21 @@ public boolean onCommand(CommandSender sender, Command command, String label, St
                     return true;
                 }
 
+                if (args[0].equals("debug") && (sender).hasPermission("picks.debug"))
```
Comment (ThePhilderbeast) [inline src/main/java/com/philderbeast/prisonpicks/PickCommands.java:49]: NOOOOOOOOOOOOOOO this should be left to be turned on in config not as a random command.

DEBUG should NEVER be turned on a live server.

File: src/main/java/com/philderbeast/prisonpicks/PickCommands.java [hunk 9]
```diff
@@ -105,7 +124,7 @@ private void sendMessage(CommandSender sender, Player receiver, String pickType,
             sender.sendMessage(Config.CHAT_FAIL_COLOR + "[" + receiver.getName() + "'s inventory was full! " + pickColour + pickType + Config.CHAT_FAIL_COLOR + " was dropped]");
         }
     }
-
+    List<String> loresList = new ArrayList<String>();
```
Comment (ThePhilderbeast) [inline src/main/java/com/philderbeast/prisonpicks/PickCommands.java:127]: MORE random class variables!

File: src/main/java/com/philderbeast/prisonpicks/PickCommands.java [hunk 10]
```diff
@@ -120,4 +139,33 @@ private boolean givePick(Player receiver, ItemStack pick)
             return false;
         }
     }
-}
\ No newline at end of file
+}
+
```
Comment (ThePhilderbeast) [inline src/main/java/com/philderbeast/prisonpicks/PickCommands.java:143]: fix your white space

File: src/main/java/com/philderbeast/prisonpicks/Pickoplenty.java [hunk 11]
```diff
@@ -25,6 +24,7 @@ public void breakBlock(BlockBreakEvent event)
         Player player = event.getPlayer();
         ItemStack item = player.getInventory().getItemInMainHand();
         Block block = event.getBlock();
+
```
Comment (ThePhilderbeast) [inline src/main/java/com/philderbeast/prisonpicks/Pickoplenty.java:27]: more random white space

File: src/main/java/com/philderbeast/prisonpicks/Pickoplenty.java [hunk 12]
```diff
@@ -56,6 +56,7 @@ public void breakBlock(BlockBreakEvent event)
                         {
                             mat = check.getBlock().getType();
                             level = Priority.getPriority(mat).level;
+
```
Comment (ThePhilderbeast) [inline src/main/java/com/philderbeast/prisonpicks/Pickoplenty.java:59]: even more random white space

File: src/main/java/com/philderbeast/prisonpicks/PrisonPicks.java [hunk 13]
```diff
@@ -1,36 +1,25 @@
...
+import org.bukkit.configuration.file.FileConfiguration;
+import org.bukkit.configuration.file.YamlConfiguration;
 import org.bukkit.plugin.java.JavaPlugin; 
 
 public class PrisonPicks extends JavaPlugin 
 {
 
     private static PrisonPicks instance; 
-
+    static File customYml;
```
Comment (ThePhilderbeast) [inline src/main/java/com/philderbeast/prisonpicks/PrisonPicks.java:12]: why are these added? the config belongs in the config option

File: src/main/java/com/philderbeast/prisonpicks/PrisonPicks.java [hunk 14]
```diff
@@ -1,36 +1,25 @@
...
+    static FileConfiguration customConfig;
+    
     @Override
     public void onLoad()
     {
+        
         Config.setConfigFolder(this.getDataFolder().getAbsolutePath()); 
         Config.reloadConfigs(); 
-
-        WorldGuardPlugin worldGuard = Util.getWorldGuard();
```
Comment (ThePhilderbeast) [inline src/main/java/com/philderbeast/prisonpicks/PrisonPicks.java:17]: why has the worldguard flag been removed?

File: src/main/java/com/philderbeast/prisonpicks/Util.java [hunk 15]
```diff
@@ -70,7 +72,6 @@ static int calculateExperienceForBlock(Block block)
                 min = 3; 
                 max = 7; 
                 break; 
-            case GLOWING_REDSTONE_ORE:
```
Comment (ThePhilderbeast) [inline src/main/java/com/philderbeast/prisonpicks/Util.java:73]: why has this been removed? remember case statements without a break fall through

File: src/main/java/com/philderbeast/prisonpicks/Util.java [hunk 16]
```diff
@@ -79,6 +80,23 @@ static int calculateExperienceForBlock(Block block)
                 min = 2; 
                 max = 5; 
                 break; 
+            case DEEPSLATE_COAL_ORE:
+                min = 0;
```
Comment (ThePhilderbeast) [inline src/main/java/com/philderbeast/prisonpicks/Util.java:84]: min 0? you can get nothing from this block?

File: src/main/java/com/philderbeast/prisonpicks/Util.java [hunk 17]
```diff
@@ -104,14 +122,22 @@ static boolean canBuild(Player player, Location location)
         WorldGuardPlugin wg = getWorldGuard(); 
         if (wg != null)
         {
-            World world = location.getWorld(); 
-            ApplicableRegionSet regions = wg.getRegionManager(world).getApplicableRegions(BukkitUtil.toVector(location)); 
+            LocalPlayer localPlayer = WorldGuardPlugin.inst().wrapPlayer(player);
```
Comment (ThePhilderbeast) [inline src/main/java/com/philderbeast/prisonpicks/Util.java:125]: why is this changed from using the custom flag?

File: src/main/java/com/philderbeast/prisonpicks/Xpick.java [hunk 18]
```diff
@@ -61,6 +65,11 @@ public void breakBlock(BlockBreakEvent event)
                             && (block.getBlock().getType() != Material.BEDROCK)
                             && (block.getBlock().getType() != Material.AIR))
                         {
+                            if (block.getWorld().getBlockAt(block).getType() == Material.EMERALD_ORE) {
```
Comment (ThePhilderbeast) [inline src/main/java/com/philderbeast/prisonpicks/Xpick.java:68]: why are you trying to send messages in the middle of finding the blocks to break?

**Review by ThePhilderbeast — COMMENTED** [review]
this needs a lot of work.

lots of things appear broken or are just realllllllly bad code.

File: src/main/java/com/philderbeast/prisonpicks/Xpick.java (hunk 18 above)
Comment (mrlubert) [reply src/main/java/com/philderbeast/prisonpicks/Xpick.java:68]: if you break a emerald with the xpick it will send the message ~ working

File: src/main/java/com/philderbeast/prisonpicks/Util.java (hunk 17 above)
Comment (mrlubert) [reply src/main/java/com/philderbeast/prisonpicks/Util.java:125]: worldguard wont register custom flag, temp work around.

File: src/main/java/com/philderbeast/prisonpicks/PrisonPicks.java (hunk 14 above)
Comment (mrlubert) [reply src/main/java/com/philderbeast/prisonpicks/PrisonPicks.java:17]: worldguard wont register custom flag, temp work around.

File: src/main/java/com/philderbeast/prisonpicks/Pick.java (hunk 3 above)
Comment (mrlubert) [reply src/main/java/com/philderbeast/prisonpicks/Pick.java:28]: for the chance to display a different break message

File: src/main/java/com/philderbeast/prisonpicks/Events.java (hunk 2 above)
Comment (mrlubert) [reply src/main/java/com/philderbeast/prisonpicks/Events.java:91]: mistakenly removed it

File: src/main/java/com/philderbeast/prisonpicks/Config.java (hunk 1 above)
Comment (mrlubert) [reply src/main/java/com/philderbeast/prisonpicks/Config.java:45]: planned on adding copper to a special mine

File: src/main/java/com/philderbeast/prisonpicks/Util.java (hunk 15 above)
Comment (mrlubert) [reply src/main/java/com/philderbeast/prisonpicks/Util.java:73]: believe 1.17.1 removed glowing_redstone_ore id

File: src/main/java/com/philderbeast/prisonpicks/Util.java (hunk 16 above)
Comment (mrlubert) [reply src/main/java/com/philderbeast/prisonpicks/Util.java:84]: mistake, meant to copy the normal coal ore

File: src/main/java/com/philderbeast/prisonpicks/Pick.java (hunk 4 above)
Comment (mrlubert) [reply src/main/java/com/philderbeast/prisonpicks/Pick.java:34]: half asleep while coding....

File: src/main/java/com/philderbeast/prisonpicks/Pick.java (hunk 3 above)
Comment (ThePhilderbeast) [reply src/main/java/com/philderbeast/prisonpicks/Pick.java:28]: this value is static though, it should be in the method, otherwise its set when its first loaded and never again

File: src/main/java/com/philderbeast/prisonpicks/Config.java (hunk 1 above)
Comment (ThePhilderbeast) [reply src/main/java/com/philderbeast/prisonpicks/Config.java:45]: and the rest of the materials? are they really more valuable then emeralds?

File: src/main/java/com/philderbeast/prisonpicks/Pick.java (hunk 5 above)
Comment (ThePhilderbeast) [reply src/main/java/com/philderbeast/prisonpicks/Pick.java:119]: so is it needed? if not remove it.

also there is a built in parseInt method, so it should be using that anyway

File: src/main/java/com/philderbeast/prisonpicks/PrisonPicks.java (hunk 14 above)
Comment (ThePhilderbeast) [reply src/main/java/com/philderbeast/prisonpicks/PrisonPicks.java:17]: nothing is more permant than a temporary fix, please do a proper fix rather then just remove it

File: src/main/java/com/philderbeast/prisonpicks/Xpick.java (hunk 18 above)
Comment (ThePhilderbeast) [reply src/main/java/com/philderbeast/prisonpicks/Xpick.java:68]: this is the worst possible place to put it, it will end up getting spammed if there is more then one in the loop and slow the overall loop down.

if this is really needed make a flag and send the message outside the loop.

File: src/main/java/com/philderbeast/prisonpicks/Util.java (hunk 16 above)
Comment (ThePhilderbeast) [reply src/main/java/com/philderbeast/prisonpicks/Util.java:84]: then use a fall through on coal so you dont repeat your self.
